pkgs/build-support/compress-drv/default.nix

```diff
@@ -65,14 +73,22 @@ let
     ext: prog:
     assert validProg ext prog;
     ''
-      find -L $out -type f -regextype posix-extended -iregex '.*\.(${formatsPipe})' -print0 \
+      find -L $out -type f -regextype posix-extended -iregex '.*\.(${formatsPipe})' ${extraFindOperands} -print0 \
         | xargs -0 -P$NIX_BUILD_CORES -I{} ${prog}
     '';
-  formatsPipe = builtins.concatStringsSep "|" formats;
+  formatsPipe = lib.concatStringsSep "|" formats;
 in
-runCommand "${drv.name}-compressed" { } ''
-  mkdir $out
-  (cd $out; ${xorg.lndir}/bin/lndir ${drv})
+runCommand "${drv.name}-compressed"
+  (
+    {
+      nativeBuildInputs = [ xorg.lndir ];
+    }
+    // (lib.optionalAttrs (drv ? pname) { inherit (drv) pname; })
+    // (lib.optionalAttrs (drv ? version) { inherit (drv) version; })
+  )
+  ''
+    mkdir $out
+    (cd $out; lndir -silent ${drv})
```

Comment:
Why silent?

Reply:
I think because the output doesn't really mean much.

Reply:
Yeah, it just prints all the directories it creates which is really noisy.